lib/vmdb/loggers/azure_logger.rb

```diff
@@ -0,0 +1,22 @@
+module Vmdb::Loggers
+  class AzureLogger < VMDBLogger
+    def initialize(*loggers)
+      super
+
+      @datetime_format = '%Y-%m-%d %H:%M:%S'
+
+      @formatter = proc do |severity, datetime, _progname, msg|
+        msg = msg.sub(/Bearer(.*?)\"/, 'Bearer [FILTERED] "')
+        msg = msg.sub(/SharedKey(.*?)\"/, 'SharedKey [FILTERED] "')
+        msg = msg.sub(/client_secret=(.*?)&/, "client_secret=[FILTERED]&")
+        "[#{datetime}] - #{severity} -- : #{msg}\n"
```

Comment:
:-1: I don't see why the log message should be completely different from other log messages.

Reply:
Does it maybe make sense to extend the the `VMDBLogger::Formatter#call` method here instead then, so we are keeping the same format, but making sure to sub what is needed?

```ruby
module Vmdb::Loggers
  class AzureLogger < VMDBLogger
    def initialize(*loggers)
      super

      # pulled from Ruby's `Logger::Formatter`, which is what it defaults to when it is `nil`
      @datetime_format = "%Y-%m-%dT%H:%M:%S.%6N "
      @formatter       = Vmdb::Loggers::AzureLogger::Formatter.new
    end

    # ...

    class Formatter < VMDBLogger::Formatter
      def call(severity, datetime, _progname, msg)
        msg = msg.sub(/Bearer(.*?)\"/, 'Bearer [FILTERED] "')
        msg = msg.sub(/SharedKey(.*?)\"/, 'SharedKey [FILTERED] "')
        msg = msg.sub(/client_secret=(.*?)&/, "client_secret=[FILTERED]&")
        super
      end
    end
  end
end
```

Reply:
I don't know how to filter the message without having to also specify the format explicitly with a logger instance. Is there a message format I can easily use?

Reply:
@djberg96 I think you were responding to what @Fryguy was asking not me (posted at roughly the same time it seems).  I think what I have supplied would do just that.